agent/agentbootstrap/bootstrap.go

```diff
@@ -244,6 +256,46 @@ func InitializeState(
 	return ctrl, nil
 }
 
+func verifyModelConfigDefaultSpace(st *state.State) error {
+	m, err := st.Model()
+	if err != nil {
+		return err
+	}
+	mCfg, err := m.Config()
+	if err != nil {
+		return err
+	}
+
+	name := mCfg.DefaultSpace()
+	if name == "" {
+		// No need to verify if a space isn't defined.
+		return nil
+	}
+
+	_, err = st.SpaceByName(name)
+	return errors.Annotatef(err, "cannot verify %s", config.DefaultSpace)
+}
+
+// maybeUpdateControllerConfigSpaces sets the JujuHASpace and
+// JujuManagementSpace to the AlphaSpaceName if either is
+// an empty string.
+func maybeUpdateControllerConfigSpaces(st *state.State) error {
```

Comment:
Now that we have changed the brief so that there is not *always* a default space configured, We shouldn't set these. Like the default space, they remain opt-in.